src/main/java/seedu/duke/commands/AddCommand.java

```diff
@@ -1,15 +1,169 @@
 package seedu.duke.commands;
 
+import seedu.duke.Duke;
+import seedu.duke.Ui;
+import seedu.duke.exceptions.DukeException;
+import seedu.duke.items.Event;
+import seedu.duke.items.Task;
+
 public class AddCommand extends Command {
-    protected static String EventDescription;
 
-    public AddCommand() {
+    protected static String itemType;
+    protected static String itemTitle;
+    protected static String itemDescription;
+    protected static String itemDate;
+
+    protected static String eventVenue;
+    protected static int eventBudget;
+
+    // Indicates if an error occurs due to the wrong format typed by the user
+    protected static boolean isCorrectFormat;
+
+    public AddCommand(String[] command, String response) {
+
+        isCorrectFormat = true;
+
+        try {
+            if (command.length == 1) {
+                throw new DukeException("Please specify what to add. ");
+            }
+            itemType = command[1];
+            if (itemType.equalsIgnoreCase("-t")) {
+                prepareTask(response);
+            } else if (itemType.equalsIgnoreCase("-e")) {
+                prepareEvent(response);
+            } else {
+                throw new DukeException("Invalid item flag entered. Please specify task '-t' or event '-e'. ");
+            }
+        } catch (DukeException e) {
+            System.out.println(e.getMessage());
+            isCorrectFormat = false;
+        }
+    }
+
+    private String retrieveItemTitle(String response) {
+        int startOfItemTitle = response.indexOf("n/") + 2;
+        int endOfItemTitle = response.indexOf("/", startOfItemTitle) - 2;
+        if (endOfItemTitle < 0) {
+            return response.trim().substring(startOfItemTitle);
+        }
+        return response.trim().substring(startOfItemTitle, endOfItemTitle);
+    }
+
+    private String retrieveItemDate(String response) {
+        int startOfItemDate = response.indexOf("d/") + 2;
+        int endOfItemDate = response.indexOf("/", startOfItemDate) - 2;
+        if (endOfItemDate < 0) {
+            return response.trim().substring(startOfItemDate);
+        }
+        return response.trim().substring(startOfItemDate, endOfItemDate);
+    }
+
+    private String retrieveEventVenue(String response) {
+        int startOfEventVenue = response.indexOf("v/") + 2;
+        int endOfEventVenue = response.indexOf("/", startOfEventVenue) - 2;
+        if (endOfEventVenue < 0) {
+            return response.trim().substring(startOfEventVenue);
+        }
+        return response.trim().substring(startOfEventVenue, endOfEventVenue);
+    }
+
```

Comment:
Code is clear in terms of retrieving the start and end index of all the various parameters needed. Good job!
It seems that `retrieveItemTitle`, `retrieveItemDate` and `retrieveEventVenue` are all very closely related. Perhaps further abstraction could make the code cleaner and shorter for future purposes eg. factor out lines such as `response.indexOf("v/") + 2`.